wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/TextDocumentService.java

```diff
@@ -403,8 +401,14 @@ public boolean canDo(ExtendedLanguageServer element) {
           @Override
           public boolean handleResult(ExtendedLanguageServer element, Hover hover) {
             if (hover != null) {
-              HoverDto hoverDto = new HoverDto(hover);
-              result.getContents().addAll(hoverDto.getContents());
+              if (hover.getContents().isLeft()) {
+                List<Either<String, MarkedString>> left = hover.getContents().getLeft();
+                result.getContents().getLeft().addAll(left);
+              } else {
+                // TODO: here we can't assume response from several servers MarkupContent will be
```

Comment:
I think we should avoid adding new TODOs because they never get done, do you think we can change the comment or implement what's needed here?

Reply:
I added TODO here because i don't find solution, what will be best to do here